Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java

```diff
@@ -216,13 +206,79 @@ public String confirmlEdit(@PathVariable("id") String id,
 				}
 			}
 		}
-
 		ConceptEntry[] arraySynonyms = new ConceptEntry[synonyms.size()];
+
 		int i = 0;
+		StringBuffer jsonStringBuilder = new StringBuffer("{");
+		jsonStringBuilder.append("\"Total\"");
+		jsonStringBuilder.append(":");
+		jsonStringBuilder.append(synonyms.size());
+		jsonStringBuilder.append(",");
+		jsonStringBuilder.append("\"synonyms\":");
+		jsonStringBuilder.append("[");
 		for (ConceptEntry syn : synonyms) {
 			arraySynonyms[i++] = syn;
+			// Appending for next element in JSON
+			if (i != 1) {
+				jsonStringBuilder.append(",");
+			}
+			buildJSONWithSynonymObject(syn, jsonStringBuilder);
 		}
-		return arraySynonyms;
+		jsonStringBuilder.append("]");
+		jsonStringBuilder.append("}");
+		return new ResponseEntity<String>(jsonStringBuilder.toString(), HttpStatus.OK);
+	}
+
+	/**
+	 * This method fetches the synonym details based on the synonym id selected
+	 * in the Add synonym table.
+	 * 
+	 * @param synonymid
+	 *            Holds the ID of a synonym
+	 * @param model
+	 *            A generic model holder for Servlet
+	 * @return synonym details
+	 * @throws JSONException
+	 */
+	@RequestMapping(method = RequestMethod.GET, value = "getConceptAddSynonyms")
+	public ResponseEntity<String> getSynonymRows(@RequestParam("synonymid") String synonymid,
+			ModelMap model) throws JSONException {
+
+		ConceptEntry synonym = conceptManager.getConceptEntry(synonymid);
+		StringBuffer jsonStringBuilder = new StringBuffer("{");
+		jsonStringBuilder.append("\"synonyms\":");
+		jsonStringBuilder.append("[");
+		buildJSONWithSynonymObject(synonym, jsonStringBuilder);
+		jsonStringBuilder.append("]");
+		jsonStringBuilder.append("}");
+		return new ResponseEntity<String>(jsonStringBuilder.toString(), HttpStatus.OK);
+	}
+	
+	
+	private void buildJSONWithoutSynonymObject(StringBuffer jsonStringBuilder, ConceptEntry syn) {
+		jsonStringBuilder.append("{");
+		jsonStringBuilder.append("\"id\":\"" + syn.getId() + "\"");
+		jsonStringBuilder.append(",");
+		jsonStringBuilder.append("\"word\":\"" + syn.getWord() + "\"");
+		jsonStringBuilder.append(",");
+		jsonStringBuilder.append("\"description\":\"" + syn.getDescription().replaceAll("\"", "'") + "\"");
+		jsonStringBuilder.append(",");
+		String pos = syn.getPos().replaceAll("\"", "'");
+		jsonStringBuilder.append("\"pos\":\"" + pos + "\"");
+		jsonStringBuilder.append("}");
+	}
+	
+	private void buildJSONWithSynonymObject(ConceptEntry synonym, StringBuffer jsonStringBuilder) {
+		jsonStringBuilder.append("{");
+		jsonStringBuilder.append("\"Id\":\"" + synonym.getId() + "\"");
+		jsonStringBuilder.append(",");
+		jsonStringBuilder.append("\"Word\":\"" + synonym.getWord() + "\"");
+		jsonStringBuilder.append(",");
+		String description = synonym.getDescription().replaceAll("\"", "'");
+		jsonStringBuilder.append("\"Description\":\"" + description + "\"");
+		jsonStringBuilder.append(",");
+		jsonStringBuilder.append("\"SynonymObject\":\"" + synonym + "\"");

```

Comment:
I have made both the attributes same. But the difference between these two methods is that For each iteration I have added a synonymObject, where as in previous method I dont have a synonym Object. And also previous method contains POS and this method does not contain pos attribute